drivers/usbdev_synopsys_dwc2/usbdev_synopsys_dwc2.c

```diff
@@ -578,15 +580,41 @@ static void _sleep_periph(const dwc2_usb_otg_fshs_config_t *conf)
     /* Unblocking STM32_PM_STOP during suspend on the stm32f446 breaks
      * while (un)blocking works on the stm32f401, needs more
      * investigation with a larger set of chips */
-#ifdef STM32_USB_OTG_CID_1x
+#if defined(STM32_USB_OTG_CID_1x)
     pm_unblock(STM32_PM_STOP);
+#elif defined(MCU_EFM32)
+    /* switch USB core clock source either to LFXO or LFRCO */
+#if defined(CPU_FAM_EFM32GG12B)
+    CMU->USBCTRL &= ~_CMU_USBCTRL_USBCLKSEL_MASK;
+    CMU->USBCTRL |= (CLOCK_LFA == cmuSelect_LFXO) ? CMU_USBCTRL_USBCLKSEL_LFXO
+                                                  : CMU_USBCTRL_USBCLKSEL_LFRCO;
+#elif defined(CPU_FAM_EFM32GG) || defined(CPU_FAM_EFM32LG)
+    CMU->CMD &= ~_CMU_CMD_USBCCLKSEL_MASK;
+    CMU->CMD |= (CLOCK_LFA == cmuSelect_LFXO) ? CMU_CMD_USBCCLKSEL_LFXO
+                                              : CMU_CMD_USBCCLKSEL_LFRCO;
+#else
+#error "EFM32 family not yet supported"
+#endif
+    pm_unblock(EFM32_PM_MODE_EM2);
 #endif
 }
 
 static void _wake_periph(const dwc2_usb_otg_fshs_config_t *conf)
 {
-#ifdef STM32_USB_OTG_CID_1x
+#if defined(STM32_USB_OTG_CID_1x)
     pm_block(STM32_PM_STOP);
+#elif defined(MCU_EFM32)
+    pm_block(EFM32_PM_MODE_EM2);
+    /* switch USB core clock source either to LFXO or LFRCO */
+#if defined(CPU_FAM_EFM32GG12B)
+    CMU->USBCTRL &= ~_CMU_USBCTRL_USBCLKSEL_MASK;
+    CMU->USBCTRL |= CMU_USBCTRL_USBCLKSEL_USHFRCO;
+#elif defined(CPU_FAM_EFM32GG) || defined(CPU_FAM_EFM32LG)
+    CMU->CMD &= ~_CMU_CMD_USBCCLKSEL_MASK;
+    CMU->CMD |= CMU_CMD_USBCCLKSEL_HFCLKNODIV;
+#else
+#error "EFM32 family not yet supported"
+#endif
```

Comment:
Maybe, `CMU_ClockSelectSet()` could be used here, too.

Reply:
Here we unfortunately need a different handling of the EFM32 families, because they use different HF clock sources for the USB peripherals.

```c
#if defined(CPU_FAM_EFM32GG12B)
    CMU_ClockSelectSet(cmuClock_USB, cmuSelect_USHFRCO);
#elif defined(CPU_FAM_EFM32GG) || defined(CPU_FAM_EFM32LG)
    CMU_ClockSelectSet(cmuClock_USB, cmuSelect_HFCLK);
#else
#error "EFM32 family not yet supported"
#endif
```

Reply:
:+1: Ok, I have tested it. With these changes suspend and wake up works perfectly. Without these changes, it didn't work correclty.